### PR TITLE
[models] Added hostname validator for Device.name #62

### LINKED ISSUE #62: Add hostname validator to Device.name
How to reproduce the issue: try to save a new device with name `test device`.
Result: netjsonconfig validation complains test device is not a valid hostname and the validation error is shown in the configuration section.
Fix: it would be better to show this validation error relating directly to the name field

### PR DESCRIPTION
**Pull request by R9295**
Implements and closes #62

### CODE REVIEW

**Comment by R9295** [comment]
@nemesisdesign  do take a look at the build-fail as flake8 complains if all the imports go in  one line, as the line exceeds the max char. limit. And if I use an alternative, isort complains.

File: django_netjsonconfig/base/device.py [hunk 1]
```diff
@@ -105,3 +109,10 @@ def get_default_templates(self):
     @classmethod
     def get_config_model(cls):
         return cls._meta.get_field('config').related_model
+
+
+# edit validators
+name_field = AbstractDevice._meta.get_field('name')
+name_field.validators = name_field.validators[:] + [
```
Comment (nemesisdesign) [inline django_netjsonconfig/base/device.py]: have you understood what's happening here? Especially the `[:]` syntax? If not, please ask!

File: django_netjsonconfig/tests/test_device.py [hunk 2]
```diff
@@ -87,3 +87,18 @@ def test_get_default_templates(self):
         self._create_config(device=d)
         self.assertEqual(d.get_default_templates().count(),
                          Config().get_default_templates().count())
+
+    def test_bad_hostnames(self):
+        bad_host_name_list = [
+            'test device',
+            'openwisp..mydomain.com',
+            'openwisp,mydomain.test'
```
Comment (nemesisdesign) [inline django_netjsonconfig/tests/test_device.py]: why the mac addresses were removed from this list? Those were a good test!

Comment (nemesisdesign) [reply django_netjsonconfig/tests/test_device.py]: ahah, just realized my comment doesn't make sense :D

**Comment by nemesisdesign** [comment]
Merged in manually in https://github.com/openwisp/django-netjsonconfig/commit/31d04aacb5604401715d6064e18b20275ffbbd51

**Pull request closed without merging**